src/module.zig

```diff
@@ -176,28 +187,112 @@ pub fn Objects(comptime T: type) type {
 
-        pub fn set(objs: *@This(), id: ObjectID, value: T) void {
+        /// This function will set all fields of an object.
+        ///
+        /// Note that this function has no tracking, and will not trigger
+        /// any internal mach functions.
+        ///
+        /// Example: core.windows.setAllRaw(id, value) will NOT trigger the Platform
+        /// to update the window's size, etc.
+        pub fn setAllRaw(objs: *@This(), id: ObjectID, value: T) void {
             const data = &objs.internal.data;
 
-            const unpacked = objs.validateAndUnpack(id, "set");
+            const unpacked = objs.validateAndUnpack(id, "setAllRaw");
+            data.set(unpacked.index, value);
+        }
+
+        /// This function will set all fields of an object.
+        ///
+        /// Note that this function also sets a bit set value
+        /// for each field, informing mach that the field has
+        /// been updated and can be passed to internals that need it.
+        ///
+        /// Example: core.windows.setAll(id, value) can trigger the Platform
+        /// to update the window's size, etc.
+        pub fn setAll(objs: *@This(), id: ObjectID, value: T) void {
+            const data = &objs.internal.data;
+
+            const unpacked = objs.validateAndUnpack(id, "setAll");
             data.set(unpacked.index, value);
+
+            if (options.track_fields)
+                objs.internal.updated.items[unpacked.index].setAll();
         }
 
-        pub fn get(objs: *@This(), id: ObjectID) ?T {
+        /// This function will set a single field of an object.
+        ///
+        /// Note that this function has no tracking, and will not trigger
+        /// any internal mach functions.
+        ///
+        /// Example: core.windows.setRaw(id, .width, value) will NOT trigger the Platform
+        /// to update the window's size.
+        pub fn setRaw(objs: *@This(), id: ObjectID, comptime field_name: anytype, value: anytype) void {
+            if (@typeInfo(@TypeOf(field_name)) != .enum_literal) @compileError("mach: invalid field name, expected `.field` enum literal, found: " ++ @typeName(@TypeOf(field_name)));
+
+            const data = &objs.internal.data;
+            const unpacked = objs.validateAndUnpack(id, "setRaw");
+
+            var current = data.get(unpacked.index);
+            @field(current, @tagName(field_name)) = value;
+
+            data.set(unpacked.index, current);
+        }
+
+        /// This function will set a single field of an object.
+        ///
+        /// Note that this function also sets a bit set value
+        /// for each object field, informing mach that the field has
+        /// been updated and can be passed to internals that need it.
+        ///
+        /// Example: core.windows.set(id, .width, value) will trigger the Platform
+        /// to update the window's size.
+        pub fn set(objs: *@This(), id: ObjectID, comptime field_name: anytype, value: anytype) void {
```

Comment:
```suggestion
        pub fn set(objs: *@This(), id: ObjectID, comptime field_name: std.meta.FieldEnum(T), value: anytype) void {
```

We can use FieldEnum here, which will return an enum for each field in `T`, guaranteeing that what we got is actually a field of the struct (and removing the need for the `hand-written `@compilerError` type check below)

Reply:
Same thing here w.r.t. comments

Reply:
`value: anytype` here and in other places can be `value: T` I think for better type safety?

Reply:
The value here is the field value, not the full object value though?